### ISSUE TITLE
Not an issue, simply a thank you

### ISSUE THREAD

**Issue by itswerkin** [e1ioan/rokuphp#21]
I just wanted to drop a note and thank you for your excellent work on rokuphp.  I've been interested in displaying our Wyzecams on our rokus for quite some time but have always failed to get things working. I took another swing at the project this weekend and after scouring the web, trying countless approaches and nearly giving up I stumbled upon your repo and was able to spin up an Ubuntu 18.04 VM, execute your installer, configure a test camera and setup ip cam viewer pro in under an hour.  Everything worked perfectly.  There's a lot of confusion on the web surrounding this subject, I really appreciate the time and effort you put into your work, thank you for sharing it!

One minor note for those who wind up here after me, the free version of ip cam viewer pro does not include the required settings to use this solution, you'll need to buy the pro version.

Thanks again.

**Comment by B1ackCr0** [comment]
I came here looking to solve this same problem.  I'm already an IP Cam Viewer Pro user, but before I dive into this can you tell if rokuphp pulls the video stream from the wyzecam full-time or only when the roku is connected to rokuphp?  And can i mix rokuphp cams and direct IP cams within IPcamPro in say a 4-way split?

**Comment by RobotNot** [comment]
Another thank you from me.  This project is awesome!  I've purchased IP Camera Viewer Pro for both of our Roku big screens, and am using rokuphp to bridge the RTSP to our UniFi Protect cameras.  Very nice!

**Comment by e1ioan** [comment]
> Another thank you from me. This project is awesome! I've purchased IP Camera Viewer Pro for both of our Roku big screens, and am using rokuphp to bridge the RTSP to our UniFi Protect cameras. Very nice!

I'm happy you like it! Happy holidays!

**Comment by crashrb** [comment]
> Another thank you from me. This project is awesome! I've purchased IP Camera Viewer Pro for both of our Roku big screens, and am using rokuphp to bridge the RTSP to our UniFi Protect cameras. Very nice!

@RobotNot Do you have any delays with your Ubiquiti cameras?  In my case it is 10-15 seconds delay on IP Camera Viewer Pro when using Roku PHP vs accessing Protect directly.  I am running the software on a Ubuntu server with an I7 processor and plenty of RAM.

**Comment by RobotNot** [comment]
> @RobotNot Do you have any delays with your Ubiquiti cameras? In my case it is 10-15 seconds delay on IP Camera Viewer Pro when using Roku PHP vs accessing Protect directly. I am running the software on a Ubuntu server with an I7 processor and plenty of RAM.

@crashrb Not at all!  Just 2 seconds behind the Protect Web UI live view.  Since IPCVP is using anonymous snapshots, the experience is like a video with very slow frame rate (about 1 image per second), but it keeps up well enough to maintain basic situational awareness.  I have Roku Ultra's on Ethernet, and RokuPHP is running on a Pi Model 3B+ POE.  Protect is hosted on a UNVR with 10 cameras, and the Rokus are displaying 4 of them.  Everything is on local network.

**Comment by crashrb** [comment]
> 

@RobotNot Interesting - I have everything on Ethernet - I may have to buy another Pi to try rather than running it on a dedicated server.  Doesn't make sense since my server is much more powerful, but that is the only thing that is different.

**Comment by SomeJ0192** [comment]
@crashrb I have a very similar setup as you.  15cams on a UNVR with rokuphp running on an olderish xeon server.  I consistently have a 10sec lag for anything on the roku/TV (which matches how long it takes for the video to load on the TV) vs viewing through the UNVR web page or UnifiProtect mobile ap.

Had the issue with ubuntu 20, and also after upgrading to 22.  Also before upgrading to a UNVR we were running the old UnifiVideo server on the same machine as rokuphp and had the same 10second lag on the TV vs directly from the camera/NVR.  Rokus are all using wired connections, and using high quality 1080p for all of my camera rtsp streams and never have any buffering or frame issues.  Just have the 10sec lag and it seems like the picture is slightly zoomed in causing part of the video all the away around the picture to be missing.  (almost as if ffmpeg is sending a slightly higher frame/picture size than my 1080p TV can support, this happens on all of our TVs but its not a horrible bother for us)

One last thought... A single ffmpeg stream/process is using 3-4% CPU on my server so I doubt it is CPU related, most likely it is a buffer setting within the ffmpeg command.  Looking through the php files, it looks like config/streamer.xml might be where some of those settings come from, as it looks like broadcast.php is referencing it for some settings.  In the streamer.xml file I changed the input -thread_queue_size from 512 to 64, killed all ffmpeg instances and reloaded the streams.  Still took 10sec for them to load up and they still had the 10sec delay.  So it doesn't appear that input buffer setting for ffmpeg is doing it, unless im missing it being set somewhere else.  Once the stream process has been started, switching to another cam and back loads much faster, however that 10sec delay in the video time stamp is still there. I did load up the same stream in VLC and it only had a 1sec lag on the camera time stamp so it is not on the UNVR side, it really feels like client side prebuffering.
